src/main/java/org/mafagafogigante/dungeon/schema/JsonObjectValidator.java

```diff
@@ -0,0 +1,34 @@
+package org.mafagafogigante.dungeon.schema;
+
+import org.mafagafogigante.dungeon.util.JsonElementSearchUtil;
+
+import com.eclipsesource.json.JsonObject;
+import com.eclipsesource.json.JsonValue;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public final class JsonObjectValidator {
+
+  private final JsonRulesFactory jsonRulesFactory;
+
+  public JsonObjectValidator(JsonRulesFactory jsonRulesFactory) {
+    this.jsonRulesFactory = jsonRulesFactory;
+  }
+
+  /**
+   * Applies json rule pairs to json object and returns validation result list.
+   */
+  public List<JsonValidationResult> applyTo(JsonObject sourceJsonObject) {
+    final List<JsonRulePair> jsonRulePairs = jsonRulesFactory.getJsonRulePairs();
+    List<JsonValidationResult> jsonValidationResults = new ArrayList<>();
+    for (JsonRulePair jsonRulePair : jsonRulePairs) {
+      List<JsonValue> sourceJsonObjectValues =
+          JsonElementSearchUtil.searchJsonValueByAttributeName(sourceJsonObject, jsonRulePair.getTargetElementName());
```

Comment:
I will only comment the first occurrence: line wraps are discouraged. We already allow for 120 character lines to prevent these issues. Introducing local variables and renaming methods and variables to shorter alternatives is a solution.

This is a minor issue, of course.